src/cpu/aarch64/gemm_arm_convolution.hpp

```diff
@@ -0,0 +1,195 @@
+
+    DNNL_DISALLOW_COPY_AND_ASSIGN(acl_resource_t);
+
+private:
+    std::unique_ptr<acl_obj_t> acl_obj_;
+
+}; // acl_resource_t
+
+struct arm_gemm_convolution_fwd_t : public primitive_t {
+    struct pd_t : public cpu_convolution_fwd_pd_t {
+        pd_t(const convolution_desc_t *adesc, const primitive_attr_t *attr,
+                const typename pd_t::base_class *hint_fwd_pd)
+            : cpu_convolution_fwd_pd_t(adesc, attr, hint_fwd_pd), acp_() {}
+
+        DECLARE_COMMON_PD_T(GEMM_IMPL_STR, arm_gemm_convolution_fwd_t,
+                USE_GLOBAL_SCRATCHPAD);
+
+        status_t init(engine_t *engine) {
+            bool ok = true && is_fwd()
+                    && set_default_alg_kind(alg_kind::convolution_direct)
+                    && expect_data_types(data_type::f32, data_type::f32,
+                            data_type::f32, data_type::f32, data_type::f32)
+                    && !has_zero_dim_memory()
+                    && attr()->has_default_values(
+                            primitive_attr_t::skip_mask_t::post_ops)
+                    && post_ops_ok();
+            if (!ok) return status::unimplemented;
+
+            auto conf_status = arm_gemm_convolution_utils::init_conf(acp_,
+                    src_md_, weights_md_, dst_md_, bias_md_, *desc(), *attr());
+            if (conf_status != status::success) return status::unimplemented;
+
+            // Number of threads in Compute Library is set by OMP_NUM_THREADS
+            // dnnl_get_max_threads() == OMP_NUM_THREADS
+            arm_compute::Scheduler::get().set_num_threads(
+                    dnnl_get_max_threads());
+
+            return status::success;
+        }
+
+        arm_conv_gemm_conf_t acp_;
+
+    protected:
+        bool post_ops_ok() const {
+            auto const &po = attr()->post_ops_;
+            auto is_eltwise
+                    = [&](int idx) { return po.entry_[idx].is_eltwise(); };
+            const int entry_idx = po.find(primitive_kind::eltwise);
+            auto act_type = po.entry_[entry_idx].eltwise.alg;
+
+            const bool eltwise_ok = (po.len() == 1) && is_eltwise(0)
+                    && arm_gemm_convolution_utils::acl_act_ok(act_type);
+
+            return eltwise_ok || (po.len() == 0);
+        }
+    };
+
+    arm_gemm_convolution_fwd_t(const pd_t *apd) : primitive_t(apd) {}
+
+    status_t create_resource(
+            engine_t *engine, resource_mapper_t &mapper) const override {
+        if (mapper.has_resource(this)) return status::success;
+
+        auto r = utils::make_unique<acl_resource_t>();
+        if (!r) return status::out_of_memory;
+
+        // Configure the resource based on information from primitive descriptor
+        auto st = r->configure(pd()->acp_);
+        if (st == status::success) { mapper.add(this, std::move(r)); }
+
+        return st;
+    }
+
+    ~arm_gemm_convolution_fwd_t() {}
+
+    typedef typename prec_traits<data_type::f32>::type data_t;
```

Comment:
Minor: we have just pushed clang-tidy checks (literally today). This one will create a hit and some others. You may need to run clang-tidy before pushing the change. You may find an instruction how to do that [here](https://github.com/densamoilov/oneDNN/blob/f768829c304ea9be50a6cdfb7147207aac813d17/rfcs/20200813-clang-tidy/README.md). Sorry for this one.

Reply:
I had a look at the link you provided, ran all my new builds with `-DDNNL_USE_CLANG_TIDY=CHECK` but nothing came up. I can see the .clang-tidy config file but couldn't find any scripts so I'm not sure if I'm missing something?

Reply:
I see that in `src/cpu/ref_convolution.hpp` we also use `typedef` instead of `using` ([link](https://github.com/oneapi-src/oneDNN/blob/f6615f305b6efa2bf2d2f573ac7e4760bbbd0d3e/src/cpu/ref_convolution.hpp#L115)). Not sure why clang-tidy doesn't fix that (summon @densamoilov).

If clang-tidy is happy, the PR should be fine.